libcst/codemod/visitors/tests/test_apply_type_annotations.py

```diff
@@ -0,0 +1,629 @@
+# Copyright (c) 2016-present, Facebook, Inc.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# pyre-strict
+
+import textwrap
+from typing import Type
+
+from libcst import parse_module
+from libcst.codemod import Codemod, CodemodContext, CodemodTest
+from libcst.codemod.visitors._apply_type_annotations import ApplyTypeAnnotationsVisitor
+from libcst.testing.utils import data_provider
+
+
+class TestApplyAnnotationsVisitor(CodemodTest):
+    TRANSFORM: Type[Codemod] = ApplyTypeAnnotationsVisitor
+
+    @data_provider(
+        (
+            (
+                """
+                def foo() -> int: ...
+                """,
+                """
+                def foo():
+                    return 1
+                """,
+                """
+                def foo() -> int:
+                    return 1
+                """,
+            ),
+            (
+                """
+                import bar
+
+                def foo() -> bar.Baz: ...
+                """,
+                """
+                def foo():
+                    return returns_baz()
+                """,
+                """
+                from bar import Baz
+
+                def foo() -> Baz:
+                    return returns_baz()
+                """,
+            ),
+            # Keep the existing `import A` instead of using `from A import B`.
+            (
+                """
+                import bar
+
+                def foo() -> bar.Baz: ...
+                """,
+                """
+                import bar
+
+                def foo():
+                    return returns_baz()
+                """,
+                """
+                import bar
+
+                def foo() -> bar.Baz:
+                    return returns_baz()
+                """,
+            ),
```

Comment:
I'm keeping the use of `GatherImports` to get and check against the imports that already exist in the source module, because otherwise it adds an extra import `from bar import Baz` and makes the return type of `foo` be `Baz`.